### PR TITLE
docs(core): fix links to devkit index page

### LINKED ISSUE #14033: Can't reach the devkit page from the docs side bar
### Documentation issue

- [ ] Reporting a typo
- [X] Reporting a documentation bug
- [ ] Documentation improvement
- [ ] Documentation feedback

### Is there a specific documentation page you are reporting?

https://nx.dev/packages/devkit
https://nx.dev/packages/devkit/index

### Additional context or description

The official docs has a broken link to the devkit index page.

### PR DESCRIPTION
**Pull request by felipeplets**
## Current Behavior

Documentation link on the sidebar menu to the DevKit package is broken.

## Expected Behavior

Documentation link is fixed.

## Related Issue(s)

Fixes #14033

### CODE REVIEW

File: docs/map.json [hunk 1]
```diff
@@ -539,7 +539,7 @@
               "id": "packages-devkit",
               "tags": ["create-your-own-plugin"],
               "file": "",
-              "path": "/packages/devkit/index",
+              "path": "/packages/devkit",
```
Comment (bcabanes) [inline docs/map.json:542]: Could you change this to:
```suggestion
            "path": "/packages/devkit/documents/index",
```

So we can keep the same intent?

**Review by bcabanes — CHANGES_REQUESTED** [review]
Thank you for the time and the contribution, I only have a minor change and it should be good to go 👍

**Comment by felipeplets** [comment]
> Thank you for the time and the contribution, I only have a minor change and it should be good to go 👍

Done. Thank you for the review @bcabanes

**Comment by bcabanes** [comment]
You will have to run `yarn documentation` again and commit the files to make the CI pass and propagate the changes to the generated documentation.

**Comment by felipeplets** [comment]
@bcabanes done.

**Review by bcabanes — APPROVED** [review]
Perfect thank you!

**Pull request merged**